CONTRIBUTING.md

```diff
@@ -0,0 +1,102 @@
+Creating Issues
+===============
+
+## :beetle: Reporting bugs
+
+reporting bugs
+
+
+## :sparkles: Requesting features
+
+requesting features
+
+
+Contributing Code to MockBukkit
+===============================
+
+## :toolbox: Software requirements
+
+To get started contributing to MockBukkit, you will need a few things.
+- [Git](https://git-scm.com/downloads)
+- Java 17 or newer ([Windows Download](https://download.oracle.com/java/17/archive/jdk-17.0.3.1_windows-x64_bin.exe)).
+
+
+## :wrench: Setting up the project
+
+The first step in contributing to MockBukkit, is to create a fork of it. You can do this by clicking the `Fork` button in the top-right of the GitHub repository.
+Once your fork is created, you will need to clone it to your local system.
+
+To do this, on GitHub, click on the green `Code` button on the homepage of your fork, then copy the provided URL.
+Next, if on Windows open up `Git Bash` and navigate to where you want to clone the repo.
+If on Linux, simply navigate to where you want to clone the repo.
+
+Once you're where you want to clone the repo, run `git clone <url>` with `<url>` being the URL you just copied, omitting the `<>`.
+
+When the previous command is finished executing, you will need to create your own branch.
+To do this, run `git checkout -b <branch name>` with `<branch name>` being the name of your new branch.
+There cannot be two branches with the same name. The recommended naming scheme for branch is `feature/my-feature` for adding something new, and `fix/my-fix` for fixing a bug.
+`my-feature`/ `my-fix` should obviously be a condensed name of what you're adding/ fixing.
+
+Now you can get to coding!
+
+
+## :black_nib: Code formatting
+
+All pull requests should adhere to our code style.
+If you're using a modern IDE like Intellij, we have a `.editorconfig` file which will tell it how code should be formatted.
+This allows you to use auto code formatting without changing more than you modified.
+If for whatever reason your IDE does not follow our style, please, do **not** commit changes that unnecessarily modify code.
+
+
+## :books: Writing documentation
+
+When overriding Bukkit methods there's no need to write your own JavaDocs.
+However, any methods or classes that are specific to MockBukkit should have JavaDocs written for them.
+
+
+## :pencil2: Writing unit tests
+
+All code you write should have unit tests to accompany it.
+The unit test class is always the same fully qualified name, but with `Test` appended to the end.
+Please create tests that cover all possible execution paths and not just one, assuming there is more than one.
```

Comment:
must be in the same package

Reply:
> must be in the same package

Well, given that the package name is part of the fully qualified name, this is already implied.